Tests/SonarScanner.MSBuild.Tasks.UnitTests/IsTestByReferenceTests.cs

```diff
@@ -59,37 +59,28 @@ public void TestReference_MultipleTestReferences_IsTest()
             ExecuteAndAssert(references, "FluentAssertions", "Resolved test reference: FluentAssertions");
         }
 
-        [DataTestMethod]
-        [DataRow("MOQ", "1.0")]
-        [DataRow("Moq", "2.0")]
-        [DataRow("MoQ", "3.0")]
-        [DataRow("moq", "4.0")]
-        // We need a different version for each test case, because AssemblyName implementation caches the instance and returns capitalization from the first usage
-        public void TestReference_TestReference_IsTest_CaseInsensitive(string name, string version) =>
-            ExecuteAndAssert(new string[] { $"{name}, Version={version}" }, name, "Resolved test reference: " + name);
-
```

Comment:
Should be reverted

Reply:
Clarification of the other thread after offline talk with @andrei-epure-sonarsource :

**sonar-dotnet & sonar-security:**
* User references MOQ or any other miss-cased reference
* MsBuild resolves the reference MOQ to a nuget package
* csc provides the references
* Roslyn sees Moq or any other resolved/correct reference, as it was compiled by the NuGet author
* 
**S4NET**
* User references MOQ or any other miss-cased reference
* MsBuild resolves the reference MOQ
* Our task is triggered with MOQ value

So we cannot use the exact same logic there.

Reply:
we'd also need an integration test with nuget resolving dependencies case insensitively to prove that it works.